libs/core/langchain_core/prompts/dict.py

```diff
@@ -0,0 +1,137 @@
+"""Dict prompt template."""
+
+import warnings
+from functools import cached_property
+from typing import Any, Literal, Optional
+
+from langchain_core.load import dumpd
+from langchain_core.prompts.string import (
+    DEFAULT_FORMATTER_MAPPING,
+    get_template_variables,
+)
+from langchain_core.runnables import RunnableConfig, RunnableSerializable
+from langchain_core.runnables.config import ensure_config
+
+
+class DictPromptTemplate(RunnableSerializable[dict, dict]):
```

Comment:
note: no longer inherits from BaseMessagePromptTemplate since thats not really how its used